src/Bicep.Core/Parser/Parser.cs

```diff
@@ -107,6 +108,15 @@ private static RecoveryFlags GetSuppressionFlag(SyntaxBase precedingNode)
             }
         }
 
+        private SyntaxBase TargetScope()
+        {
+            var keyword = ExpectKeyword(LanguageConstants.TargetScopeKeyword);
+            var assignmentToken = this.Expect(TokenType.Assignment, b => b.ExpectedCharacter("="));
```

Comment:
>Expect [](start = 39, length = 6)

You should do WithRecovery here to parse the assignment token. Otherwise, the targetScope keyword won't be colorized until someone types `=`.

Reply:
I need that behavior for IntelliSense.

---
In reply to: [516891660](https://github.com/Azure/bicep/pull/771#discussion_r516891660) [](ancestors = 516891660)

Reply:
Oof, yeah, not sure how I missed that... It'll probably cause unhandled exceptions to be thrown as well.

Reply:
We still have the outer recovery that would handle the exceptions, so that would have been ok.